KalturaMediaServer/src/com/kaltura/media/server/api/KalturaWebServicesServer.java

```diff
@@ -42,11 +44,12 @@ public Endpoint addService(IWebService service)
 	}
 
 	public void shutdown() {
-		Set<String> keys = services.keySet();
-		for(String key : keys)
+		Iterator<Map.Entry<String, Endpoint>> serviceIter = services.entrySet().iterator();
```

Comment:
This is the only place that we remove from this collection, and therefore there is no need to synchronize it. But I don't care enough, you can keep this code.